docs/how-to/use-rewards-api.md

```diff
@@ -0,0 +1,135 @@
+
+A [Swagger interface is also available](../reference/rest.md#enable-the-rest-api-service). Use the [`--rest-api-docs-enabled`](../reference/cli/index.md#rest-api-docs-enabled) command line option to enable the web interface.
+
+:::
+
+## Limitations
+
+The following limitations apply:
+
+- The rewards API currently supports the `altair` fork upgrade and later.
+
+- The rewards API relies on state and block data to retrieve the reward information, meaning you'll receive limited data if the beacon node being queried is not an archive node.
+
+- Querying from a beacon node in `prune` storage mode only returns information for the the `finalized` block, and nothing before that.
+
+## Impact of data storage modes
+
+:::tip
+
+You can change the [`data-storage-mode`](../reference/cli/index.md#data-storage-mode) without re-initializing your database.
+
+You can change the frequency that states are stored by specifying [`data-storage-archive-frequency`](../reference/cli#data-storage-archive-frequency), but it will only affect the state storage from the time that the change has been made. It will also directly impact the amount of disk space required by Teku.
+
+:::
+
+Consider using a beacon node with `archive` mode storage if you frequently call the rewards API on finalized data. However, this may produce slow results due to having to replay blocks due to the infrequent storage of states on disk (every 2048 slots by default).
+
+## Examples
+
+Query all the rewards from the block currently at head.
+
+<!--tabs-->
+
+# Example
+
+```bash
+curl http://localhost:5051/eth/v1/beacon/rewards/blocks/head |jq
+```
+
+# Result
+
+```json
+{
+  "execution_optimistic": false,
+  "finalized": false,
+  "data": {
+    "proposer_index": "555552",
+    "total": "1217168",
+    "attestations": "0",
+    "sync_aggregate": "1217168",
+    "proposer_slashings": "0",
+    "attester_slashings": "0"
+  }
+}
+```
+
+<!--/tabs-->
+
+Query all the sync committee rewards from the block at head for validator index 1.
+
+<!--tabs-->
+
+# Example
+
+```bash
+curl -X POST \
+     -H 'accept: application/json' \
+     -H 'Content-Type: application/json' \
+     http://localhost:5051/eth/v1/beacon/rewards/sync_committee/head \
+     -d '["1"]' |jq
+```
+
+# Result
+
+```json
+{
+  "execution_optimistic": false,
+  "finalized": false,
+  "data": [
+      {
+        "validator_index": "1",
+        "reward": "16778"
+      }
+    ]
+  }
+}
+```
+
+<!--/tabs-->
+
+Query attestation rewards from epoch 204644, just for validator index 0.
+
+<!--tabs-->
+
+# Example
+
+```bash
+curl -X POST \
+     -H 'accept: application/json' \
+     -H 'Content-Type: application/json' \
+     http://localhost:5051/eth/v1/beacon/rewards/attestations/204648 \
+     -d '["0"]' |jq
+```
+
+# Results
+
+```json
+ {
+  "execution_optimistic": false,
+  "finalized": true,
+  "data": {
+    "ideal_rewards": [...],
```

Comment:
This shows as invalid json which is technically true, it was from my example where i did `...` instead of listing all the objects under it... I'm happy to leave it invalid, just highlighted red in review so figured i'd comment on it...

Reply:
Cool, I'll leave the elipses, we often use it if the results are to long to display.